### PR TITLE
KOGITO-2959 Remove redundant smallrye dependency from runtimes

### PR DESCRIPTION
**Pull request by evacchi**

### CODE REVIEW

**Comment by evacchi** [comment]
re: https://github.com/kiegroup/kogito-runtimes/pull/642/files/b94b82e1a354895f2816f937644f7b8d2834745a#r465776360

a fix may be that we do not put it in <dependenycManagement> but we explicitly depend on it using the version property. This way importing the bom won't conflict with other boms. WDYT? @radtriste @ricardozanini 

see also https://issues.redhat.com/browse/KOGITO-2726

File: pom.xml [hunk 1]
```diff
@@ -120,7 +120,6 @@
     <version.io.quarkus>1.6.0.Final</version.io.quarkus>
     <version.io.restassured>4.3.0</version.io.restassured>
     <version.io.restassured.springboot>3.3.0</version.io.restassured.springboot>    <!-- Manually override to 3.3.0 because Spring boot uses rest-assured-common@3.3.0, otherwise error: The type io.restassured.common.mapper.TypeRef cannot be resolved. It is indirectly referenced from required .class files -->
-    <version.io.smallrye.reactive>2.1.1</version.io.smallrye.reactive>
```
Comment (ricardozanini) [inline pom.xml:123]: Just a reminder to add it back! #642

**Comment by ricardozanini** [comment]
We need the version since `smallrye-reactive-messaging-http` is not available on Quarkus.

**Comment by evacchi** [comment]
folded this inside #682

**Pull request closed without merging**